Review followup: PR #21520 — GH#21499: fix: resolve SC2027/SC2086 quoting bug in _cmd_enable_systemd printf

## Unaddressed review bot suggestions

PR #21520 was merged with unaddressed review bot feedback. Each comment
below includes its file path, line number, a direct link to the inline
review comment, and a `diff` fence with the code context the bot was
flagging. Resolved and outdated threads are filtered out via GitHub's
GraphQL review-thread state. Read the relevant lines, decide whether
the suggestion is correct, and either apply the fix or close this issue
with a wontfix rationale.

**Source PR:** https://github.com/marcusquinn/aidevops/pull/21520

---

### You are the triager (worker-is-triager rule)

This issue is **auto-created from review bot output** and dispatched
directly to you. Review bots can be wrong: hallucinated line refs, false
premises about codebase structure, template-driven sweeps without
measurements (see GH#17832-17835 for prior art and `prompts/build.txt`
section 6a). **Do not assume the bot is correct.** Verify before acting.

You must end in exactly one of three outcomes — no fourth "hand it back
to the human" path exists. Humans approve decisions; they do not re-do
analysis.

#### Outcome A — Premise falsified → close the issue

1. Read the cited `file:line` (listed under *Files to modify* below).
2. If the bot's claim is factually wrong (file doesn't exist at that
   line, function doesn't behave as described, "auto-generated" section
   isn't actually auto-generated, etc.), **close the issue** with a
   comment in this shape:

   > **Premise falsified.** \<what the bot claimed\>. \<what the code
   > actually shows, with a `file:line` citation or one-line quote\>.
   > Not acting.

   No PR. No further dispatch. The closing comment trains the next
   session reading this thread and the noise filter.

#### Outcome B — Premise correct + fix is obvious → implement and PR

1. Verify the bot's premise as above.
2. Read the Worker Guidance section below, open a worktree, implement.
3. Open a PR with `Resolves #<this-issue-number>` in the body
   (use THIS issue's number, not the source PR's) so merge auto-closes it.
4. Follow the normal Lifecycle Gate (brief, tests, review-bot-gate,
   merge, postflight).

#### Outcome C — Premise correct but approach is a genuine judgment call

Only use this path if you reach it after Outcomes A and B don't apply:
the bot's finding is real, but the fix requires a decision that is
architectural, policy, breaking-change, or otherwise genuinely outside
what you can resolve autonomously. In that case, post a **decision
comment** with exactly these fields:

- **Premise check:** one line, confirming the finding is real.
- **Analysis:** 2-4 bullets on the trade-offs.
- **Recommended path:** the option you would take if the decision were
  yours, with rationale.
- **Specific question:** the single decision the human needs to make
  (yes/no or pick-one, not open-ended).

Then apply `needs-maintainer-review` and stop. The human wakes up to a
ready-to-approve recommendation, not a blank task.

> **Ambiguity about scope or style is not Outcome C.** Per
> `prompts/build.txt` "Reasoning responsibility", the model does the
> thinking and delivers a recommendation. Only escalate what is genuinely
> a maintainer-only decision.

### Worker Guidance

**Files to modify:**

- `.agents/scripts/auto-update-helper.sh:883`

**Implementation steps (Outcome B path):**

1. Read the `diff` block under each inline comment below — it shows the
   exact code the bot was flagging. Open the file only if you need
   surrounding context beyond what the diff tail shows.
2. Read the bot's full comment below the diff — it contains the rationale
   and any suggested change.
3. Verify the premise before implementing (see Outcome A). If the premise
   is wrong, switch to Outcome A instead of burning iterations trying to
   satisfy a wrong suggestion.
4. If multiple comments target the same file, group your edits into one
   logical commit.
5. Run `shellcheck` / `markdownlint-cli2` / project tests as appropriate.

**Verification:**

- Open the new PR with `Resolves #<this-issue>` so this followup is auto-closed on merge.
- If the bot's suggestion was incorrect, close this issue with a Outcome A comment — do not open a no-op PR.

### Inline comments

#### gemini-code-assist on `.agents/scripts/auto-update-helper.sh:883`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/21520#discussion_r3153069613)

```diff
 
-	# shellcheck disable=SC1078,SC1079,SC2027,SC2086  # systemd ExecStart wraps a single-quoted shell command; bash interpolates "${script_path}" into the systemd directive value, embedded single+double quotes are intentional
+	# shellcheck disable=SC1078,SC1079  # multi-line printf with single quotes inside a double-quoted string; ${script_path} kept inside outer double quotes for safe expansion
 	printf '%s' "[Unit]
 Description=aidevops auto-update
 After=network.target
 
 [Service]
 Type=oneshot
 KillMode=process
-ExecStart=/bin/bash -lc '"${script_path}" check'
+ExecStart=/bin/bash -lc '${script_path} check'
```

> ![high](https://www.gstatic.com/codereviewagent/high-priority.svg)
> 
> The current fix correctly addresses the shell quoting issue within the `printf` command itself (resolving SC2027/SC2086), but it introduces a regression for paths containing spaces in the generated `systemd` unit file. 
> 
> When `systemd` executes the command `bash -lc '${script_path} check'`, it passes the string `${script_path} check` as a single argument to `bash -c`. If `${script_path}` contains spaces, `bash` will perform word splitting on that string and fail to find the script (it will try to execute the first part of the path). 
> 
> To fix this, you should wrap `${script_path}` in escaped double quotes so that the resulting service file contains a properly quoted path for the shell to interpret.
> 
> ```suggestion
> 	ExecStart=/bin/bash -lc '\"${script_path}\" check'
> ```

### PR review summaries

#### gemini-code-assist[bot] review summary

[View review](https://github.com/marcusquinn/aidevops/pull/21520#pullrequestreview-4187554002)

> ## Code Review
> 
> This pull request updates the `auto-update-helper.sh` script to modify the formatting of the systemd service's `ExecStart` directive and update associated ShellCheck suppressions. A review comment identifies a regression where the removal of quotes around the script path would cause the command to fail if the path contains spaces. A suggestion was provided to use escaped double quotes to ensure the path is correctly handled by the shell during execution.
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.13.9 automated scan.

--- a/.agents/scripts/auto-update-helper.sh
+++ b/.agents/scripts/auto-update-helper.sh
@@ -872,7 +872,7 @@ _cmd_enable_systemd() {
 
 	mkdir -p "${SYSTEMD_SERVICE_DIR}"
 
-	# shellcheck disable=SC1078,SC1079  # multi-line printf with single quotes inside a double-quoted string; ${script_path} kept inside outer double quotes for safe expansion
+	# shellcheck disable=SC1078,SC1079  # multi-line printf with single quotes inside a double-quoted string; \"${script_path}\" expands to "path" in the service file, preserving space-safe quoting
 	printf '%s' "[Unit]
 Description=aidevops auto-update
 After=network.target
@@ -880,7 +880,7 @@ After=network.target
 [Service]
 Type=oneshot
 KillMode=process
-ExecStart=/bin/bash -lc '${script_path} check'
+ExecStart=/bin/bash -lc '\"${script_path}\" check'
 TimeoutStartSec=120
 Nice=10
 IOSchedulingClass=idle

